eng/Publishing.props

```diff
@@ -31,125 +35,72 @@
+               $(ArtifactsPackagesDir)**\Microsoft.NET.Workload.Mono.ToolChain.*.nupkg" />
+  </ItemGroup>
+  <ItemGroup Condition="$(PackageRID.StartsWith('win-'))">
+    <PackagesToPublishFromThisJob
+      Include="$(ArtifactsPackagesDir)**\VS.Redist.Common.NetCore.*.nupkg" />
+  </ItemGroup>
 
+  <ItemGroup>
+    <PackagesToPublishFromThisJob Include="@(OtherPackages)" Exclude="@(PackagesToPublishFromThisJob)" Condition="'$(PublishAllPackages)' == 'true' or '$(DotNetBuildSourceOnly)' == 'true'" />
+    <PackagesToSkipFromThisJob Include="$(ArtifactsPackagesDir)**\*.nupkg" Exclude="@(PackagesToPublishFromThisJob)" />
     <!--
-      Assuming all symbol packages ship and can be found by turning .nupkg => .symbols.nupkg, find
-      them. Don't check for missing symbol packages here: some nupkgs don't have them for valid
-      reasons, such as the VS insertion packages that transport MSIs. Symbol package validation
-      will check for symbol completeness with file-by-file granularity rather than looking for
-      missing symbols.nupkg files: https://github.com/dotnet/arcade/issues/2499.
-
-      Handles several conventions:
-      * NonShipping packages have symbol nupkgs that are Shipping.
-      * Shipping packages have symbol packages in a "symbols" subdirectory.
+      In non-VMR builds, we can skip publishing RID-agnostic packages entirely when we're not the lane that is supposed to publish them.
+      In VMR builds, we need to publish them for upstack jobs to consume.
     -->
-    <PotentialSymbolNupkgToPublishFile
-      Include="
-        @(NupkgToPublishFile->Replace('\NonShipping\', '\Shipping\')->Replace('.nupkg', '.symbols.nupkg'));
-        @(NupkgToPublishFile->Replace('\NonShipping\', '\NonShipping\symbols\')->Replace('.nupkg', '.symbols.nupkg'));
-        @(NupkgToPublishFile->Replace('\Shipping\', '\Shipping\symbols\')->Replace('.nupkg', '.symbols.nupkg'))" />
-
-    <SymbolNupkgToPublishFile Include="@(PotentialSymbolNupkgToPublishFile->Distinct()->Exists())" />
-
-    <!-- Packages -->
-    <Artifact Include="@(NupkgToPublishFile)"
-              IsShipping="$([System.String]::new('%(Identity)').Contains('\Shipping\'))"
-              PublishFlatContainer="false" />
-    <Artifact Include="@(SymbolNupkgToPublishFile)" PublishFlatContainer="false" />
-
-    <!-- Blob storage -->
-    <Artifact Include="@(UploadToBlobStorageFile)"
-              Exclude="@(NupkgToPublishFile);@(SymbolNupkgToPublishFile)"
-              IsShipping="$([System.String]::new('%(Identity)').Contains('\Shipping\'))"
-              ChecksumPath="%(FullPath).sha512" />
-    <Artifact Include="@(WorkloadsVSInsertionFile)"
-              SubBlobFolder="workloads/"
-              IsShipping="$([System.String]::new('%(Identity)').Contains('\Shipping\'))" />
+    <Artifact Update="@(PackagesToSkipFromThisJob)" SkipPublish="true" Condition="'$(DotNetBuildOrchestrator)' != 'true'" />
```

Comment:
I'm not sure I understand why you set the `SkipPublish` metadata here instead of just removing them items?

Reply:
I would prefer to use standard msbuild functionality here (item `Remove` action) than relying on custom metadata. Not just because I question the use-case for the `SkipPublish` metadata in general when we stop importing Sign.props in Publish.proj (hehe, that's what you get when you introduce new extension points in API) but because I find the control harder to follow with this.